ci/build.sh

```diff
@@ -1,15 +1,20 @@
-#!/bin/sh -ex
+#!/bin/sh -e
 # This script is used by travis.yml and docker.sh
 
 if [ -n "${SPECIAL}" ]; then
-  sh ci/${SPECIAL}.sh
+  sh ci/"${SPECIAL}".sh
 else
-  mkdir -p ${BUILDDIR:-build}
-  cd ${BUILDDIR:-build}
-  ${CMAKE_PREFIX:-} cmake ${CCACHE_LOC:-..} ${CMAKE_PARAMS:-}
-  # 4 threads seems a reasonable default for Travis
-  ${CMAKE_PREFIX:-} cmake --build . ${BUILDEXTRAFLAGS:-} -- -j4
-  # Warning: Rare random failures when running with j4.
-  test "${RUN_TESTS:-1}" -eq "1" && ctest --output-on-failure -j1
+  JOBS=$(getconf _NPROCESSORS_ONLN 2>/dev/null) # Linux
+  [ -z "$JOBS" ] && JOBS=$(getconf NPROCESSORS_ONLN) # FreeBSD
+  [ -z "$JOBS" ] && JOBS=$(ksh93 -c 'getconf NPROCESSORS_ONLN') # Solaris
+  [ -z "$JOBS" ] && JOBS=2
+  test "$JOBS" -gt 7 && JOBS=$((JOBS - 1)) # Leave one free for mails, browsing, etc
```

Comment:
Why 7?

Reply:
"random" number:
- When you have only 1 core you don't want to keep 1 free ;-)
- Even when you have 2 cores you don't want to keep 50% free!
- I have 8 cores (4*2) and I do want to keep one free. Theoretically it could cost me 12,5% time, but usually my CPU is busy 95% that way. So that last free core is not really free anyway.
Loosing 5% speed so that all other programs keep running fine seems ok.

Somewhere in there is the sweet spot and I picked my personal setup.

Reply:
It looks strange to say "Leave one free for mails, browsing, etc" and then use your personal value. Unless it's your personal script, that is. Maybe we should use 63 since I have access to a 64 CPU machine? :slightly_smiling_face:

How about letting the user decide with e.g. `JOBS=7 the-script` or `the-script -j 7` instead?

Reply:
Ok now you got me searching for the optimal number... and of course there are as many opinions as there are people. The range seems to be cores-1 up to cores*2,5. However the speedup when using more jobs then cores seems quite small.

I think for a "small" project like ccache this is not mission critical / can be improved later. Using jobs = cores is a huge improvement over cores = 1.

Reply:
OK, right. But what do you think of my suggestion? Maybe the user _does_ want to use only one CPU sometime because of $reasons?

Reply:
As I just discovered in the other finding this script is actually only for CI. Users would normally not use it?
Maybe through docker, but that would mean we need to pass -j through all the other scripts too.

Nevertheless using the environment variable could be as trivial as prefixing the first line with `[ -z "$JOBS" ] && `. I can certainly do that.